dotcom-rendering/src/components/DynamicSlowMPU.tsx

```diff
@@ -128,216 +156,206 @@ const ColumnOfThree50_Ad50 = ({
 	);
 };
 
-type MPUProps = {
-	groupedTrails: DCRGroupedTrails;
-	containerPalette?: DCRContainerPalette;
-	showAge?: boolean;
-	adIndex: number;
-};
-
-const MPUSlice = ({
+/**
+ * DynamicSlowMPU
+ *
+ * This container only allows big and standard cards (from groupedTrails)
+ *
+ * Layout is dynamic deopending on the number of big cards. You're only
+ * allowed to have 2 or 3 big cards. If you pass 1 a standard card will
+ * get promoted to make two bigs. If you pass more than 3 bigs then the
+ * extras will get demoted to standard.
+ *
+ * If you pass no bigs at all the top slice will not render and a special
+ * 3 column layout is used for the remaining slice.
+ *
+ */
+export const DynamicSlowMPU = ({
 	groupedTrails,
 	containerPalette,
 	showAge,
 	adIndex,
-}: MPUProps) => {
-	let layout:
-		| 'noBigs'
+	renderAds,
+}: Props) => {
+	let firstSliceLayout:
+		| undefined
 		| 'twoBigs'
 		| 'twoBigsFirstBoosted'
 		| 'twoBigsSecondBoosted'
 		| 'threeBigs';
-	let bigCards: DCRFrontCard[] = [];
-	let standardCards: DCRFrontCard[] = [];
-	switch (groupedTrails.big.length) {
-		case 0: {
-			standardCards = groupedTrails.standard;
-			layout = 'noBigs';
-			break;
-		}
-		case 1: {
-			// If there is only one big item a standard item is promoted such that
-			// there is always at least two
-			const promoted = groupedTrails.standard.slice(0, 1);
-			bigCards = [...groupedTrails.big, ...promoted];
-			standardCards = groupedTrails.standard.slice(1);
-			layout = 'twoBigs';
-			break;
+	let firstSliceCards: DCRFrontCard[] = [];
+	if (groupedTrails.big.length === 1) {
+		// If there is only one big item a standard item is promoted such that
+		// there is always at least two
+		const promoted = groupedTrails.standard.slice(0, 1);
+		firstSliceCards = [...groupedTrails.big, ...promoted];
+		firstSliceLayout = 'twoBigs';
```

Comment:
What happens if we receive 1 big and boosted? Should the layout become `twoBigsFirstBoosted`. Not sure if that's even possible, just asking!

Reply:
Ignore this, I hadn't had coffee ☕